build-system/task-runner/amp-task-runner.js

```diff
@@ -7,28 +7,34 @@ const argv = require('minimist')(process.argv.slice(2));
 const commander = require('commander');
 const esprima = require('esprima');
 const fs = require('fs-extra');
+const omelette = require('omelette');
 const os = require('os');
 const path = require('path');
 const {
   updatePackages,
   updateSubpackages,
 } = require('../common/update-packages');
-const {cyan, green, magenta, red} = require('kleur/colors');
+const {cyan, green, magenta, red, yellow} = require('kleur/colors');
 const {isCiBuild} = require('../common/ci');
 const {log} = require('../common/logging');
 
 /**
  * @typedef {Function & {
  *  description: string,
- *  flags?: Object,
+ *  flags?: {[flag: string]: string},
  * }}
  */
 let TaskFuncDef;
 
 /**
- * Special-case constant that indicates if `amp --help` was invoked.
+ * Special-case constants that indicates if `amp --help`, `amp --compgen`, or
+ * `amp --setup-auto-complete` were invoked.
  */
 const isHelpTask = argv._.length == 0 && argv.hasOwnProperty('help');
+const isCompGen = !!argv.compgen;
```

Comment:
It's what `bash-completion` (or equivalent for other terminals) passes when calling `amp --compbash --compgen ${componentNumber}`
You can look at the generated `$HOME/.amp/completion.sh` file for the full call

The task runner needs to know this because the task runner is what provides the autocomplete information. I don't mind renaming the variable, but the flag name itself is outside of our control